internal/transport/transport_test.go

```diff
@@ -1991,17 +1991,33 @@ func (s) TestClientDecodeHeaderStatusErr(t *testing.T) {
 				Fields: []hpack.HeaderField{
 					{Name: "content-type", Value: "application/grpc"},
 					{Name: "grpc-status", Value: "0"},
+					{Name: ":status", Value: "200"},
 				},
 			},
 			// no error
 			wantStatus: status.New(codes.OK, ""),
 		},
+		{
+			name: "missing content-type header",
+			metaHeaderFrame: &http2.MetaHeadersFrame{
+				Fields: []hpack.HeaderField{
+					{Name: "grpc-status", Value: "0"},
+					{Name: ":status", Value: "200"},
+				},
+			},
+			// no error
+			wantStatus: status.New(
+				codes.Unknown,
+				"unexpected HTTP status code received from server: 200 (OK); missing content-type",
```

Comment:
We don't want to see an error about an unexpected 200 when the only actual problem is missing content-type.

Reply:
Done - makes total sense